environment/src/test/java/com/sequenceiq/environment/experience/liftie/LiftieConnectorListClustersTest.java

```diff
@@ -70,71 +76,41 @@ void testInvocationBuilderProviderShouldCreateCallForExecutionBasedOnTheWebTarge
         getUnderTest().listClusters(TEST_ENV_NAME, TEST_TENANT, null, null);
 
         verify(getMockRetryableWebTarget(), times(ONCE)).get(any());
         verify(getMockRetryableWebTarget(), times(ONCE)).get(getMockInvocationBuilder());
-        verify(getMockResponseReader(), never()).read(any(), any(), any());
+        verify(getMockResponseReader(), times(ONCE)).read(LIFTIE_CLUSTER_ENDPOINT_PATH, getMockResponse(), ListClustersResponse.class);
     }
 
     @Test
-    void testWhenCallExecutionReturnsNullThenThenEmptyResponseShouldReturn() {
-        when(getMockRetryableWebTarget().get(getMockInvocationBuilder())).thenReturn(null);
-
-        ListClustersResponse result = getUnderTest().listClusters(TEST_ENV_NAME, TEST_TENANT, null, null);
-
-        assertNotNull(result);
-
-        assertNull(result.getClusters());
-
-        assertNotNull(result.getPage());
-        assertEquals(0, result.getPage().getTotalPages());
-    }
-
-    @Test
-    void testWhenCallExecutionReturnsResponseButItThrowsRuntimeExceptionThenEmptyResponseShouldReturn() {
+    void testWhenCallExecutionReturnsResponseButItThrowsRuntimeExceptionThenIllegalStateExceptionIsThrown() {
         when(getMockRetryableWebTarget().get(getMockInvocationBuilder())).thenReturn(getMockResponse());
         doThrow(RuntimeException.class).when(getMockResponseReader()).read(any(), any(), any());
 
-        ListClustersResponse result = getUnderTest().listClusters(TEST_ENV_NAME, TEST_TENANT, null, null);
-
-        assertNotNull(result);
+        IllegalStateException thrown = assertThrows(IllegalStateException.class, () -> getUnderTest().listClusters(TEST_ENV_NAME, TEST_TENANT, null, null));
 
-        assertNull(result.getClusters());
-
-        assertNotNull(result.getPage());
-        assertEquals(0, result.getPage().getTotalPages());
+        assertTrue(thrown.getCause() instanceof RuntimeException);
     }
 
     @Test
-    void testWhenResponseReaderReturnsEmptyResultWhichThrowsIllegalStateExceptionThenEmptyResponseShouldReturn() {
+    void testWhenResponseReaderReturnsEmptyResultWhichThrowsIllegalStateExceptionThenIllegalStateExceptionIsThrown() {
         when(getMockRetryableWebTarget().get(getMockInvocationBuilder())).thenReturn(getMockResponse());
         when(getMockResponseReader().read(LIFTIE_CLUSTER_ENDPOINT_PATH, getMockResponse(), ListClustersResponse.class)).thenReturn(Optional.empty());
 
-        ListClustersResponse result = getUnderTest().listClusters(TEST_ENV_NAME, TEST_TENANT, null, null);
+        IllegalStateException thrown = assertThrows(IllegalStateException.class, () -> getUnderTest().listClusters(TEST_ENV_NAME, TEST_TENANT, null, null));
 
-        assertNotNull(result);
-
-        assertNull(result.getClusters());
-
-        assertNotNull(result.getPage());
-        assertEquals(0, result.getPage().getTotalPages());
+        assertTrue(thrown.getCause() instanceof RuntimeException);
```

Comment:
assertThrows will throw its exception when not the expected type of exception comes at the result of the executed method, therefore here one should make the assertion on the message instead of the actual type.
Also since the "thrown" field is already a subclass of the RuntimeException class this assertion has no point in the first place

Reply:
The same goes for the other places

Reply:
Fixed